qa-pipelines/qa-pipeline.yml.erb

```diff
@@ -0,0 +1,268 @@
+<%
+
+# ERB processing section
+
+def bundle_url_from_version(version)
+  case version
+    when true    then ''      # Use bucket version
+    when false   then ''      # Don't install (should not be reached)
+    when /:\/\// then version # URL
+    else "https://((s3-config-bucket-sles)).s3.amazonaws.com/((s3-config-prefix-sles))#{ERB::Util.url_encode(version)}.zip"
+  end
+end
+
+%>
+---
+# Concourse pipeline to deploy and upgrade CAP
+
+resource_types:
+- name: github-status
+  type: docker-image
+  source:
+    repository: resource/github-status
+    tag: release
+
+resources:
+# USB sidecar archive for postgres
+- name: s3.pg-sidecar
+  type: s3
+  source:
+    endpoint: ((s3-config-endpoint))
+    access_key_id: ((s3-config-access-key))
+    secret_access_key: ((s3-config-secret-key))
+    bucket: ((s3-config-bucket-sles))
+    regexp: ((s3-config-prefix-sidecars))cf-usb-sidecar-postgres-(.*)\.tgz$
+
+# USB sidecar archive for MySQL
+- name: s3.mysql-sidecar
+  type: s3
+  source:
+    endpoint: ((s3-config-endpoint))
+    access_key_id: ((s3-config-access-key))
+    secret_access_key: ((s3-config-secret-key))
+    bucket: ((s3-config-bucket-sles))
+    regexp: ((s3-config-prefix-sidecars))cf-usb-sidecar-mysql-(.*)\.tgz$
+
+# CI helper scripts repository
+- name: ci
+  type: git
+  source:
+    uri: ((src-ci-repo))
+    branch: ((src-ci-branch))
+    paths:
+    - qa-pipelines/*
+    - sample-apps/*
+
+# SCF bundle to install from (upgrade target, if upgrading)
+- name: s3.scf-config-sles
+  type: s3
+  source:
+    endpoint: ((s3-config-endpoint))
+    access_key_id: ((s3-config-access-key))
+    secret_access_key: ((s3-config-secret-key))
+    bucket: ((s3-config-bucket-sles))
+    regexp: ((s3-config-prefix-sles))scf-sle-(.*)\.zip$
+
+# Pool resource with kube cluster information
+- name: pool.kube-hosts
+  type: pool
+  source:
+    uri: ((kube-pool-repo))
+    private_key: ((kube-pool-key))
+    branch: ((kube-pool-branch))
+    pool: ((kube-pool-pool))
+
+# GitHub commit status
+<% if status_reporting %>
+- name: status.src
+  type: github-status
+  source:
+    repo: ((src-repo))
+    access_token: ((github-access-token))
+<% end %>
+
+jobs:
+<% %w(SA HA).each do |avail| %>
```

Comment:
Looping like this is great, as it allows us to halve the template size, but it's unclear to me why we'd want to deploy pipelines which don't have both jobs already (or pipelines which have no jobs if neither of the flags are set), which can happen by being able to set the ha and sa flags individually

Reply:
OK, two separate points:

- Scaled HA is not something our pipeline has ever preferred since we switched to using the HA flag. If we had a build which the default scaling of `HA=true` introduced breakages, we would then want to set the scaled HA sizing parameters accordingly and merge a change in our pipeline manifest to use `SCALED_HA` instead of `HA` here. The last time we deployed with the `SCALED_HA` setting was roughly a year ago, when we modified the qa-pipeline file to set `HA: true` instead: https://github.com/SUSE/cf-ci/commit/921e31e30d6a2d11c6d868f32c4c61ad5614d07c . For our template I think we can ignore this, and then in the future if we need to use `SCALED_HA` again for some reason to get more fine-grained control of the scaling, we'd merge the desired parameter changes into the cf-deploy-upgrade-common.sh file accordingly and modify the template to change HA to SCALED_HA

- The pipeline as defined by our manifest deploys two jobs, one SA and one HA. The flags you added allow you to turn one or the other job on or off, but as it is, all jobs are paused by default and we only unpause the one we want to run builds from. Sometimes I'll deploy a pipeline, unpause the SA job and run a build from that, and then later unpause the HA job and run that one too. So I don't think we gain anything from turning these jobs on or off at the template level, and furthermore, it introduces the potential to accidentally deploy a pipeline in which both jobs are disabled (which effectively means the pipeline does nothing)

Reply:
Okay, 79d89eb0a6336da9734e06b59fdb31a9a919c816 changed to drop the SA/HA flags. The scaled HA stuff is basically dead code that won't be reached right now.